src/jit/x64/emit.dasc

```diff
@@ -1534,6 +1636,48 @@ void MVM_jit_emit_primitive(MVMThreadContext *tc, MVMJitCompiler *compiler, MVMJ
         | mov WORK[dst], TMP2;
         break;
     }
+    case MVM_OP_scgethandle: {
+        MVMint16 dst = ins->operands[0].reg.orig;
+        MVMint16 sc  = ins->operands[1].reg.orig;
+        | mov TMP5, aword WORK[sc];
+        | cmp_repr_id TMP5, TMP6, MVM_REPR_ID_SCRef;
+        | je >1;
+        | throw_adhoc "Must provide an SCRef operand to scgethandle"
+        |1:
+        | mov ARG1, TC;
+        | mov ARG2, SCREF:TMP5;
+        | callp &MVM_sc_get_handle;
+        | mov WORK[dst], RV;
+        break;
+    }
+    case MVM_OP_scobjcount: {
+        MVMint16 dst = ins->operands[0].reg.orig;
+        MVMint16 sc  = ins->operands[1].reg.orig;
+        | mov TMP5, aword WORK[dst];
+        | cmp_repr_id TMP5, TMP6, MVM_REPR_ID_SCRef;
+        | je >1;
+        | throw_adhoc "Must provide an SCRef operand to scobjcount"
+        |1:
+        | mov ARG1, TC;
+        | mov ARG2, SCREF:TMP5;
+        | callp &MVM_sc_get_object_count;
+        | mov WORK[dst], RV;
+        break;
+    }
+    case MVM_OP_setobjsc: {
+        MVMint16 dst = ins->operands[0].reg.orig;
+        MVMint16 sc  = ins->operands[1].reg.orig;
+        | mov TMP5, WORK[sc];
+        | cmp_repr_id TMP5, TMP6, MVM_REPR_ID_SCRef;
+        | je >1;
+        | throw_adhoc "Must provide an SCRef operand to setobjsc"
+        |1:
+        | mov ARG1, TC;
+        | mov ARG2, aword WORK[dst];
+        | mov ARG3, SCREF:TMP5;
+        | callp &MVM_sc_set_obj_sc
```

Comment:
The semicolon that is missing here is unnecessary for DynASM, but essential for proper C syntax highlighting and formatting :smile: